frontend/reducers/document.js

```diff
@@ -1,40 +1,96 @@
 'use strict'
 
 import * as Constants from 'lib/constants'
+import * as LocalStorage from 'lib/local-storage'
 import * as Types from 'actions/actionTypes'
 
 const initialState = {
+  dirty: false,
+  loadedFromLocalStorage: false,
   local: null,
   peers: null,
   remote: null,
   remoteStatus: Constants.STATUS_IDLE,
   validationErrors: {}
 }
 
+// Boolean fields are a bit special. Any required field that hasn't
+// been touched by the user in the UI should generate a validation error,
+// except for Boolean fields, as these don't have a "neutral" state.
+// You can start editing a document and see a Boolean field set to
+// "false" and that may be exactly how you want it, but at that point the
+// field doesn't exist in the document store yet because there wasn't a
+// change event. To get around this, when we create our local copy of
+// the document we check for any Boolean fields in the collection
+// that aren't in the document object and set those to `false`.
```

Comment:
I don't think so. That is purely visual, in case we want to display a neutral state to the user, which I believe we don't.

> The indeterminate state is visual only. The checkbox is still either checked or unchecked as a state. That means the visual indeterminate state masks the real value of the checkbox, so that better make sense in your UI!